gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Dataset.java

```diff
@@ -35,111 +30,108 @@
  *
  * <p>Objects of this class are immutable. Operations that modify the dataset like {@link #update}
  * return a new object. To get a {@code Dataset} object with the most recent information use
- * {@link #reload}.
+ * {@link #reload}. {@code Dataset} adds a layer of service-related functionality over
+ * {@link DatasetInfo}.
  * </p>
  */
-public final class Dataset {
+public final class Dataset extends DatasetInfo {
 
-  private final BigQuery bigquery;
-  private final DatasetInfo info;
+  private static final long serialVersionUID = -4272921483363065593L;
 
-  private static class TablePageFetcher implements PageImpl.NextPageFetcher<Table> {
+  private final BigQueryOptions options;
+  private transient BigQuery bigquery;
 
-    private static final long serialVersionUID = 6906197848579250598L;
+  public static final class Builder extends DatasetInfo.Builder {
 
-    private final BigQueryOptions options;
-    private final Page<TableInfo> infoPage;
+    private final BigQuery bigquery;
+    private final DatasetInfo.BuilderImpl infoBuilder;
 
-    TablePageFetcher(BigQueryOptions options, Page<TableInfo> infoPage) {
-      this.options = options;
-      this.infoPage = infoPage;
+    private Builder(BigQuery bigquery) {
```

Comment:
In resource manager and storage, I've been using the package-private constructor (e.g. `new Blob(options, new BlobInfo.BuilderImpl(BLOB_INFO))`, and to test the builder itself I did something [a little round about](https://github.com/ajkannan/gcloud-java/commit/75dbbf8263003e5df370c74bd2a96942b36d35ff#diff-81eda95fc1298d0b4b24c8561147a3d5R86).  If it makes testing way more convenient, we can keep it though.